Framework/CurveFitting/src/Algorithms/SplineBackground.cpp

```diff
@@ -95,14 +95,16 @@ void SplineBackground::exec() {
 
   /* this is the data to be fitted */
   int j = 0;
-  for (MantidVec::size_type i = 0; i < Y.size(); ++i) {
+  for (MantidVec::size_type i = 0; i < yInputVals.size(); ++i) {
     if (isMasked && masked[i])
       continue;
     gsl_vector_set(x, j,
-                   (isHistogram ? (0.5 * (X[i] + X[i + 1]))
-                                : X[i])); // Middle of the bins, if a histogram
-    gsl_vector_set(y, j, Y[i]);
-    gsl_vector_set(w, j, E[i] > 0. ? 1. / (E[i] * E[i]) : 0.);
+                   (isHistogram
+                        ? (0.5 * (xInputVals[i] + xInputVals[i + 1]))
+                        : xInputVals[i])); // Middle of the bins, if a histogram
```

Comment:
You can avoid this ternary and the `isHistogram` variable by using `inWS->points(i)` to get the middle of the bins.